### PR TITLE
Fix flakey test_subscription timeouts

### LINKED ISSUE #8670: [Flaky Test] diem-json-rpc::stream_rpc::subscription_types::tests::test_subscription

### PR DESCRIPTION
**Pull request by CapCap**
## Motivation
A timeout of 50ms may be too small here, and causing the test to be flakey. Raising this value should ensure this doesn't happen

Fixes #8670 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?
yes

## Test Plan
this is the test

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Changing JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

**Review by dimroc — APPROVED** [review]

File: json-rpc/src/stream_rpc/subscription_types.rs [hunk 1]
```diff
@@ -325,15 +325,15 @@ mod tests {
         // These should have no delay
         let expected =
             serde_json::json!({"jsonrpc": "2.0", "id": 1010101, "result":{"value": 0}}).to_string();
-        let next = timeout(10, receiver.recv(), "message 0")
+        let next = timeout(1_000, receiver.recv(), "message 0")
```
Comment (bmwill) [inline json-rpc/src/stream_rpc/subscription_types.rs:328]: what's the reason why we're even using timeouts in these tests? if we don't ever expect these to fail

Comment (CapCap) [reply json-rpc/src/stream_rpc/subscription_types.rs:328]: so that they fail fast in the event they are failing

**Review by bmwill — DISMISSED** [review]

**Pull request closed without merging**